src/client.rs

```diff
@@ -10,7 +10,76 @@ use std::collections::HashSet;
 //     GameOver,
 // }
 
+pub struct Scoreboard {
+    // timer: u32,
+    found_mines: u32,
+    total_mines: u32,
+    message: String,
+    position: (f32, f32),
+}
+
+impl Scoreboard {
+    pub fn new(total_mines: u32) -> Scoreboard {
+        Scoreboard {
+            found_mines: 0,
+            total_mines,
+            message: String::from(""),
+            position: (0., 0.),
+        }
+    }
+
+    pub fn draw(&self, board_size: (f32, f32)) {
+        const TEXT_WIDTH_OFFSET: f32 = 30.;
+        const TEXT_HEIGHT_OFFSET: f32 = 40.;
+        const PADDING: f32 = 5.;
+
+        draw_rectangle_lines(
+            self.position.0 + PADDING,
+            self.position.1 + PADDING,
+            board_size.0 - PADDING * 2.,
+            board_size.1 - PADDING * 2.,
+            2.,
+            BLACK,
+        );
+        draw_text(
+            format!("Mines: {}/{}", self.found_mines, self.total_mines).as_str(),
+            self.position.0 + TEXT_WIDTH_OFFSET,
+            self.position.1 + TEXT_HEIGHT_OFFSET,
+            30.,
+            BLACK,
+        );
+        draw_text(
+            // self.message.as_str(),
+            "Hello, Minesweeper!",
+            self.position.0 + TEXT_WIDTH_OFFSET + 400.,
+            self.position.1 + TEXT_HEIGHT_OFFSET,
+            30.,
+            BLACK,
+        );
+    }
+
+    pub fn update_marked_mines(&mut self, is_marked: bool) {
+        if is_marked {
+            self.found_mines += 1;
+        } else {
+            self.found_mines -= 1;
+        }
+    }
+}
+
+struct BoardOffset {
+    x: f32,
+    y: f32,
+}
+
+impl BoardOffset {
+    fn new(heigth: f32) -> BoardOffset {
+        BoardOffset { x: 0., y: heigth }
+    }
+}
+
 pub struct Client {
+    scoreboard: Scoreboard,
```

Comment:
there is a classical approach to "how to draw a bunch of elements on screen"

u do it with a vecor of TDrawable, where TDrawable implements draw method, which for field will draw field and for scoreboard will draw scoreboard (this method has multiple downsides, needs some research, so no changes required here)